packages/unmock-node/src/serialize/index.ts

```diff
@@ -1,10 +1,70 @@
+import * as http from "http";
+import * as readable from "readable-stream";
+import { ISerializedRequest } from "unmock-core";
+
+/**
+ * Network serializers
+ */
 import _CompositeDeserializer from "./deserializer/composite";
 import _FormDeserializer from "./deserializer/form";
 import _JSONDeserializer from "./deserializer/json";
 import _CompositeSerializer from "./serializer/composite";
 import _FormSerializer from "./serializer/form";
 import _JSONSerializer from "./serializer/json";
 
+class BodySerializer extends readable.Transform {
+  public static async fromIncoming(incomingMessage: http.IncomingMessage) {
+    const serializer = new BodySerializer();
+    await readable.pipeline(incomingMessage, serializer);
+    return serializer.body;
+  }
+
+  private body: string = "";
+
+  // TODO Encoding
+  public _transform(chunk: Buffer, _: string, done: () => void) {
+    this.body += chunk.toString();
+    this.push(chunk);
+    done();
+  }
+}
+
+/**
+ * @param interceptedRequest
+ */
+export const serializeRequest = async (
+  interceptedRequest: http.IncomingMessage,
+): Promise<ISerializedRequest> => {
+  const { host } = interceptedRequest.headers;
+
+  if (!host) {
+    throw new Error("No host");
+  }
+
+  const { method, url } = interceptedRequest;
+  if (!url) {
+    throw new Error("Missing url.");
+  }
+  if (!method) {
+    throw new Error("Missing method");
+  }
```

Comment:
Could maybe put all of these checks in a `validateRequestOrThrow(interceptedRequest);` to simplify the function?